### PR TITLE
feat: GKE entity provider / processor

### PR DESCRIPTION
**Pull request by mclarke47**
Signed-off-by: Matthew Clarke <mclarke@spotify.com>

## Hey, I just made a Pull Request!

TODO

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mclarke47** [comment]
@freben does this look right? I'm not too familiar with providers/processors

File: plugins/catalog-backend-module-gcp/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "express": "^4.18.1",
+    "express-promise-router": "^4.1.0",
+    "node-fetch": "^2.6.7",
+    "winston": "^3.2.1",
+    "yn": "^4.0.0"
+  },
+  "devDependencies": {
+    "@backstage/cli": "workspace:^",
+    "@types/supertest": "^2.0.8",
+    "msw": "^0.47.0",
```
Comment (freben) [inline plugins/catalog-backend-module-gcp/package.json:37]: we're at `^0.49.0` now

File: plugins/catalog-backend-module-gcp/src/processors/GkeClusterProcessor.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      };
+      emit({
+        type: 'entity',
+        entity,
+        location,
+      });
+      return true;
+    }
+  }
+
```
Comment (freben) [inline plugins/catalog-backend-module-gcp/src/processors/GkeClusterProcessor.ts:81]: I see some `prettier` errors in here, gonna want to turn on auto formatting on save

File: plugins/catalog-backend-module-gcp/src/processors/GkeClusterProcessor.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        metadata: {
+          annotations: {
+            // TODO add necessary params
+            // [ACCOUNTID_ANNOTATION]: accountId,
+            // [ARN_ANNOTATION]: describedCluster.cluster.arn || '',
+            [ANNOTATION_KUBERNETES_API_SERVER]: cluster.endpoint || '',
+            [ANNOTATION_KUBERNETES_API_SERVER_CA]: '',
+            [ANNOTATION_KUBERNETES_AUTH_PROVIDER]: 'google',
+          },
+          name: cluster.name ?? "",
```
Comment (freben) [inline plugins/catalog-backend-module-gcp/src/processors/GkeClusterProcessor.ts:65]: gonna have to have a name either way

File: plugins/catalog-backend-module-gcp/src/processors/GkeClusterProcessor.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        spec: {
+          type: 'kubernetes-cluster',
+          owner: 'unknown',
+        },
+      };
+      emit({
+        type: 'entity',
+        entity,
+        location,
+      });
```
Comment (freben) [inline plugins/catalog-backend-module-gcp/src/processors/GkeClusterProcessor.ts:77]: Hm so ... this one requires you to register one location for every cluster, and it only emits one corresponding actual cluster resource. The use case gets a little odd. Should it instead auto discover clusters perhaps? Maybe with some allowlist / rejectlist / filter function if more fine grained control is needed? Or perhaps even skip the processor entirely and just lean in on the provider below

**Review by freben — COMMENTED** [review]
some early review check

File: plugins/catalog-backend-module-gcp/src/processors/GkeClusterProcessor.ts (hunk 4 above)
Comment (mclarke47) [reply plugins/catalog-backend-module-gcp/src/processors/GkeClusterProcessor.ts:77]: Yeah, I'm not sure what the best way to do this is? Should the provider emit an account location or a GCP project and the processor emitted all the clusters?

Comment (freben) [reply plugins/catalog-backend-module-gcp/src/processors/GkeClusterProcessor.ts:77]: I'm not sure what the use case (as seen from the end user's perspective) you are fulfilling with this one, so not sure what's best

**Comment by freben** [comment]
Ping @mclarke47 on this one! Still relevant? Can I help somehow?

**Comment by mclarke47** [comment]
@freben I might reimplement, still not 100% how I want to do this

**Pull request closed without merging**